src/som/compiler/Parser.java

```diff
@@ -1486,7 +1485,10 @@ private ExpressionNode nestedBlock(final MethodBuilder builder) throws ProgramDe
       blockPattern(builder);
     }
 
-    String outerMethodName = stripColons(builder.getOuterBuilder().
+    if (builder.getNextMethod() == null) {
+      throw new java.lang.NullPointerException("The outer context of a block should never be null");
```

Comment:
Why fully qualified?
And also why `NullPointer`? Isn't `IllegalState` or something more appropriate?